pages/search.js

```diff
@@ -44,45 +44,47 @@ export default function Search({ data }) {
   const [filteredUsers, setFilteredUsers] = useState([]);
   const [inputValue, setInputValue] = useState(username || keyword || "");
 
-  let results = [];
-
   useEffect(() => {
     if (username) {
       setNotFound(username);
     }
   }, [username]);
 
-  const filterData = (value) => {
-    const valueLower = value.toLowerCase();
-    const terms = valueLower.split(",");
+  const filterData = useCallback(
```

Comment:
By wrapping `fitlerData` in `useCallback` hook, we ensure that it's the same function between re-renders(until dependency change).